### PR TITLE
Update to 2.40.1 and build with OpenSSL 3 [skip ci]

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Update to 2.40.1 and build with OpenSSL 3.

Depends on:
* https://github.com/AnacondaRecipes/curl-feedstock/pull/22

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -42,21 +42,21 @@ requirements:
...
-    - libcurl   # [unix]
-    - expat     # [unix]
-    - libiconv  # [osx]
-    - openssl   # [unix]
-    - pcre2     # [unix]
+    - curl 7.88.1  # [unix]
+    - libcurl 7.88.1  # [unix]
+    - expat 2.4.9  # [unix]
+    - libiconv 1.16  # [osx]
+    - openssl 3.0.8  # [unix]
```
Comment (ELundby45) [inline recipe/meta.yaml:46]: Should this be using the pinning in the cbc?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -7,22 +7,22 @@ package:
...
+    sha256: 4fcce89aa84c79be6641fee54cf05c590c7d967734536770bf6299cae26c8a51  # [not win]
 
   - url: https://github.com/git-for-windows/git/releases/download/v{{ version }}.windows.1/PortableGit-{{ version }}-64-bit.7z.exe  # [win64]
     folder: .  # [win64]
-    sha256: dbf63703f7a37a374591450f1b1466b83ceccb724067521786bf8c5f69ed3ced  # [win64]
+    sha256: 9e1d819aef3284420adf6d923b0d4865254bd403641d915975e49ddea1e7cdf9  # [win64]
   - url: https://github.com/git-for-windows/git/releases/download/v{{ version }}.windows.1/PortableGit-{{ version }}-32-bit.7z.exe  # [win32]
     folder: .  # [win32]
-    sha256: 95e198aaf6e6455910facfc522f7981934822fb362f6605375751b8a9c62db55  # [win32]
+    sha256: e1360e94cb292862fb023018578a1029022a09278b160f7264c6dc444f65c9ca  # [win32]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: win32 source code can be removed as we do not build it already

File: recipe/meta.yaml (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:46]: If I'm not mistaken, we decided to be explicit instead of implicit for OpenSSL. @ryanskeith am I right?

**Review by ViridianMelody — APPROVED** [review]
Agree with Serhii's points. Otherwise, looks good to me!

**Review by varlackc — COMMENTED** [review]
It might be a good idea to remove the `abs.yaml` file before merging the package to defaults.

File: recipe/meta.yaml (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:46]: Alright, we just had a talk and we'll use `{{ openssl }}` and rely on aggregate.

**Comment by ryanskeith** [comment]
We are in a special case mode here as OpenSSL3 is rolling out.  We may indeed need to leave in abs.yaml files and others to ensure that other team members don't overwrite our build numbers.

**Comment by danpetry** [comment]
@ELundby45 @varlackc @skupr-anaconda do you have any remaining changes required to be addressed, or are you able to give this an approval if not?

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Review by ELundby45 — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**